### PR TITLE
Initialize boilerplate and implement necessary cryptography

### PR DESCRIPTION
**Pull request by yegortimoshenko**
This exposes API described in https://github.com/Holo-Host/chaperone-key-manager/blob/ced9150e84a3a852a00efd5f04c9c4aebc155be6/pkg/README.md.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,6 +1,77 @@
...
+
+    #[wasm_bindgen]
+    pub fn verify(&self, message: &[u8], signature_bytes: &[u8]) -> Result<bool, JsValue> {
+        let signature = Signature::from_bytes(signature_bytes).map_err(into_js_error)?;
+        Ok(self.0.verify(message, &signature).is_ok())
+    }
+
+    #[wasm_bindgen]
+    pub fn agent_id(&self) -> Result<String, JsValue> {
+	let hcid = HCID_CODEC.encode(&self.0.public.to_bytes())
```
Comment (yegortimoshenko) [inline src/lib.rs]:
```suggestion
	HCID_CODEC.encode(&self.0.public.to_bytes()).map_err(into_js_error)
```

File: src/lib.rs [hunk 2]
```diff
@@ -1,6 +1,77 @@
...
+	    .expect("Failed to encode public key");
```
Comment (yegortimoshenko) [inline src/lib.rs]:
```suggestion
```

File: src/lib.rs [hunk 3]
```diff
@@ -1,6 +1,77 @@
...
+	Ok(hcid)
```
Comment (yegortimoshenko) [inline src/lib.rs]:
```suggestion
```

File: .gitignore [hunk 4]
```diff
@@ -1,4 +1,6 @@
+dist/
 node_modules/
 pkg/
 result
 target/
+*~
```
Comment (yegortimoshenko) [inline .gitignore]: https://gist.github.com/subfuzion/db7f57fff2fb6998a16c#file-global-gitignore-md

Comment (mjbrisebois) [reply .gitignore]: Thanks, I didn't know about global ignoring.  I'll start using it

**Pull request merged**